### ISSUE TITLE
[Question] Debugging "Error: Failure when receiving data from the peer"

### ISSUE THREAD

**Issue by hjkelly** [getinsomnia/insomnia#809]
- Insomnia Version: Version 5.14.9 (5.14.9.1895)
- Operating System: Mac Sierra 10.12.6

**tl;dr:** Can I see the exact request Insomnia is sending so I can reproduce this via cURL and figure out what our internal tooling doesn't like (maybe some header Insomnia adds)?

## Details

I work on a team of ~20 developers, and my boss and I both use Insomnia to tinker with the API/platform we build. This is deployed via a Kubernetes-esque orchestration framework, and about a month ago, both of us witnessed a total breakage of our Insomnia. Our auth endpoints still work, but the rest all fail with "Error: Failure when receiving data from the peer". Meanwhile, this extremely basic request (GET + auth header) succeeds using cURL and Postman.

Given that no other issues include the words "peer", I'm guessing this was triggered by some change in our common libraries or the tooling that connects our VMs to one another. Interestingly, the successful cURL and failed Insomnia requests both show as a `200` response in Datadog and within the VM's logs.

The Question: Since I refuse to go back to Postman quietly, do you have any tips for debugging this on the client side? Maybe there's a debug mode, or a way I can see the full headers being sent (perhaps reproducing in cURL and tinkering with individual headers)? Or whether any data is received before the error occurs? Obviously I'll have some server-side debugging to turn on as well, but I'd appreciate anything that can help me see more on the client side.

**Comment by gschier** [comment]
Have you taken a look at the Timeline response tab? It shows the same information (plus a bit more) that `curl -v` does.

**Comment by hjkelly** [comment]
@gschier Thanks, that helps! Now I see this output, and that it _is_ getting the server's 200 properly, and even getting a significant amount of data before it gives up. (This whole time, I thought it was just failing to hear back from the server at all.)

I wonder if this problem could be related to cookies. I'm not adding cookies in my headers at all; is there any way to clear cookies, or turn off them being sent automatically?

**Comment by hjkelly** [comment]
@gschier Thanks to your help earlier, I've narrowed this down to Insomnia for Mac. Both Insomnia's Timeline and `curl -v` show: `SSLRead() return error -9806`. Interestingly, cURL still manages to show the response's body while Insomnia gives up and shows the error mention in my original post. 

I was able to get rid of that error with cURL by reinstalling it with the Homebrew OpenSSL, but that doesn't fix Insomnia. I found #302 but the steps to work around this aren't clear to me. I also found #504, but trying `GET https://pasteurid.pasteur.fr/password/` failed whether SSL validation was toggled on or off.

As another data point, hitting these URLs from Insomnia for Ubuntu seems to work fine! My work laptop has to stay on 10.12, but I'll try this from my personal Mac when I get home.

I'm hitting the limits of my knowhow (not much experience with Mac or Electron app compilation), but if you can point me in the right direction, I'm willing to try anything. Like I said, Postman is not an option. ;)

**Comment by rafaelbpa** [comment]
For some reason, restarting my mac solved. It is not an Insomnia problem (although the error message could be better). Sometimes, for reasons which I do not know, my mac keeps losing some partial internet functionality (e.g. browsing works fine, steam works fine etc, BUT discord app does not connect, also npm and yarn downloads/updates and git stream do not work too). I always try a lot of different things (like renew internet DHCP concession and also change to a new wi-fi) before to restart my mac, but none of them solves. Unfortunately restarting is the option which works for me. Just writing this, because someone can be with the same problem as me.

**Comment by SamuelAubin** [comment]
For anyone stuck with this issue (7 years later), it just came up for me. 
Ended up being a datadog agent listening on the same port, and I believe insomnia was hitting the agent as opposed to my api. It was tricky to debug because i could access from my web browser but not insomnia, making me think it was an insomnia issue. All good now.
